fairscale/optim/oss.py

```diff
@@ -286,10 +288,20 @@ def clip_grad_norm(
             total_norm = total_norm ** (1.0 / norm_type)
 
         clip_coef = torch.tensor(max_norm, dtype=total_norm.dtype, device=total_norm.device) / (total_norm + 1e-6)
+
+        if skip_nan_grads and torch.isnan(clip_coef):
+            return total_norm
+
+        assert not torch.isnan(clip_coef), (
```

Comment:
by default we assume that normalizing by a NaN is most probably the sign that something is broken, better signal it early